src/symtab/symbol_table.hpp

```diff
@@ -131,11 +131,28 @@ class SymbolTable {
         return parent ? parent->name() : "None";
     }
 
-    std::vector<std::shared_ptr<Symbol>> get_variables(syminfo::NmodlType with,
-                                                       syminfo::NmodlType without);
+    /**
+     * get variables
+     *
+     * \param with variables with properties. 0 matches everything
+     * \param without variables without properties. 0 matches nothing
+     *
+     * The two different behaviors for 0 depend on the fact that we get
+     * get variables with ALL the with properties and without ANY of the
+     * without properties
+     */
+    std::vector<std::shared_ptr<Symbol>> get_variables(
+        syminfo::NmodlType with = syminfo::NmodlType::empty,
+        syminfo::NmodlType without = syminfo::NmodlType::empty);
```

Comment:
So with this, call to `get_variables()` returns now empty vector? (or, it is supposed to return all variables?)

Reply:
it returns all the variables because in get_variables there is a call to get_variables_with_properties(..., all = true). Already tested with the python interface

Reply:
it is a little confusing but for get_variables_with_properties(..., true) you get all the variables with NmodlType = 0 and nothing with NmodlType = -1. If you use get_variables_with_properties(..., false) exactly the opposite.

Reply:
since returning no variable is trivial I switched the behavior to the most useful one